### PR TITLE
Added lab and Cypress tests

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -6,6 +6,8 @@ workflows:
     jobs:
       - cypress/run:
           start: npm start
+          spec:
+            - "cypress/integration/index.test.js"
```
Comment (mikeboyle) [inline .circleci/config.yml:10]: We haven't put this in quotes in the past. Should we have?

File: README.md [hunk 2]
```diff
@@ -1,6 +1,28 @@
...
+
 [![Pursuit Logo](https://avatars1.githubusercontent.com/u/5825944?s=200&v=4)](https://pursuit.org)
 
-# [Assignment Title]
+## Tasks
+
+Build an app that has the following functionality:
+
+- Users can create a new Todo by typing text into an `input` inside a `form` and pressing enter
+- Shows a list of todos with id `#todos` underneath that input using instances of a `Todo` component
```
Comment (mikeboyle) [inline README.md:10]: `that input using instances` ... I'm not sure I follow what that means

File: README.md [hunk 3]
```diff
@@ -1,6 +1,28 @@
...
+- Users can delete an existing Todo by pressing an 'x' `button` next to it
+
+Implement lifecycle methods to fire the created/deleted notifications:
```
Comment (mikeboyle) [inline README.md:13]: Should there be a little more context here, first e.g.: `When a todo is created or deleted, a pop-up "toast" will appear to notify the user that the action has been completed.`

File: README.md [hunk 4]
```diff
@@ -1,6 +1,28 @@
...
+
+- Use `componentDidMount` in `Todo` to show a `"success"` toast with the text `New todo added: {todo}`
+- Use `componentWillUnmount` in `Todo` to show an `"error"` toast with the text `Todo deleted: {todo}`
```
Comment (mikeboyle) [inline README.md:16]: 👍  Thank you for giving the specific guidance here on which lifecycle methods to use!

File: README.md [hunk 5]
```diff
@@ -1,6 +1,28 @@
...
+
+For the toasts notifications, use the [react-toastify](https://github.com/fkhadra/react-toastify) library that has already been installed with this lab.
```
Comment (mikeboyle) [inline README.md:18]: 👍  Nice, thanks for already importing the library!

File: README.md [hunk 6]
```diff
@@ -1,6 +1,28 @@
...
+See its README.md for example usage.
+
+![todos react app in use](./todosLifecycleAppGif.gif)
+
+### Bonus
+
+Use `componentDidUpdate` in `App` to show a toast with the text `Remaining todos: {count}` whenever the count of todos changes.
```
Comment (mikeboyle) [inline README.md:25]: 👍  love this

File: cypress/integration/index.test.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
+describe("Index", () => {
+  it("adds a todo and toast when submitted", () => {
+    cy.visit("http://localhost:3000");
+    cy.get("input").type("say hello");
+    cy.get("form").submit();
+    cy.get("#todos").contains("say hello");
+    cy.contains("New todo added: say hello");
```
Comment (mikeboyle) [inline cypress/integration/index.test.js:7]: Agree with the strategy of `cy.contains` in real life when we know that the toast has actually been built and we want to verify its existence in the proper user-facing way.

For this assignment, though, should we do more here to assert that this text appears in a toast and not just in a random div? What is the lift for that? (I don't know the React toast library and how well it plays with cypress -- is there a class or similar we can select on?)

File: src/App.js [hunk 8]
```diff
@@ -1,13 +1,18 @@
...
-        <h1>Hello, world!</h1>
-      </main>
-    </div>
-  );
+class App extends React.Component {
+  render() {
+    return (
+      <div className="app">
+        <ToastContainer />
+        Give me something to do?!
```
Comment (mikeboyle) [inline src/App.js:12]: 😄  lol

File: README.md [hunk 9]
```diff
@@ -1,6 +1,28 @@
...
+- Shows a list of todos with id `#todos` underneath that input using instances of a `Todo` component
+- Users can delete an existing Todo by pressing an 'x' `button` next to it
+
+Implement lifecycle methods to fire the created/deleted notifications:
+
+- Use `componentDidMount` in `Todo` to show a `"success"` toast with the text `New todo added: {todo}`
+- Use `componentWillUnmount` in `Todo` to show an `"error"` toast with the text `Todo deleted: {todo}`
+
+For the toasts notifications, use the [react-toastify](https://github.com/fkhadra/react-toastify) library that has already been installed with this lab.
+See its README.md for example usage.
```
Comment (mikeboyle) [inline README.md:19]: Can we add something like this:
>You might find [the react-toastify gist](https://github.com/fkhadra/react-toastify#the-gist) or the [demo playground](https://fkhadra.github.io/react-toastify/introduction/) very helpful in getting started.

**Review by mikeboyle — COMMENTED** [review]
Looking great! A couple of things, nothing that's a deal breaker! 🥇

File: .circleci/config.yml (hunk 1 above)
Comment (JoshuaKGoldberg) [reply .circleci/config.yml:10]: Oh I don't think it matters, I can just remove them

File: README.md (hunk 2 above)
Comment (JoshuaKGoldberg) [reply README.md:10]: `that input` = the input from before, so rewording: `Shows (a list of todos with id `#todos`) (underneath the input) using instances of a `Todo` component`

File: cypress/integration/index.test.js (hunk 7 above)
Comment (JoshuaKGoldberg) [reply cypress/integration/index.test.js:7]: Totally doable - they have "stable" (guaranteed not to change across non-major releases) class names so we can select on that.

File: README.md (hunk 2 above)
Comment (mikeboyle) [reply README.md]: Thank you! It makes sense to me now and the previous phrasing also makes sense to me now. Reading comprehension fail from earlier.

File: cypress/integration/bonus.test.js [hunk 10]
```diff
@@ -4,7 +4,7 @@ describe("Bonus", () => {
     cy.get("input").type("say hello");
     cy.get("form").submit();
     cy.get("#todos").contains("say hello");
-    cy.contains("New todo added: say hello");
+    cy.get(".Toastify__toast--success").contains("New todo added: say hello");
```
Comment (mikeboyle) [inline cypress/integration/bonus.test.js:7]: This ups the challenge by requiring the toast to have the right `type` property. I'm fine with this (it is also what the gif in the README shows) but in this case we should probably make that requirement clear in the README. I.e., when the todo is created, the toast should be type `success`, when the todo is deleted, the toast should be type `error`, for the bonus use the default type (no type specified).

The alternative is to only look for some kind of base `.Toastify` class if that exists on the element

**Review by mikeboyle — APPROVED** [review]
LGTM! One final small thing on the instructions, which I can take on if you'd rather merge this and move forward. 🚀

File: cypress/integration/bonus.test.js (hunk 10 above)
Comment (JoshuaKGoldberg) [reply cypress/integration/bonus.test.js:7]: Is this not clear enough?

> - Use `componentDidMount` in `Todo` to show a `"success"` toast with the text `New todo added: {todo}`
> - Use `componentWillUnmount` in `Todo` to show an `"error"` toast with the text `Todo deleted: {todo}`

I'll defer to you on phrasing, whatever you think is appropriate 😄

Comment (mikeboyle) [reply cypress/integration/bonus.test.js:7]: Ah, I think I missed that. Thank you!

**Pull request merged**